elasticsearch.md

```diff
@@ -577,221 +577,480 @@ _Example: the simplest example of an allow block._
+      jwt_auth:
+        name: "jwt_provider_1"
+        groups: ["viewer"]
+```
+
+### Kibana-related rules
+
+#### `kibana_access`
+
+`kibana_access: ro`
+
+Enables the minimum set of actions necessary for browsers to use Kibana.
+
+This "macro" rule allows the minimum set of actions necessary for a browser to use Kibana. This rule allows a set of actions towards the designated kibana index \(see `kibana_index` rule - defaults to ".kibana"\), plus a stricter subset of read-only actions towards other indices, which are considered "data indices".
+
+The idea is that with one single rule we allow the bare minimum set of index+action combinations necessary to support a Kibana browsing session.
+
+Possible access levels:
+
+* `ro_strict`: the browser has a read-only view on Kibana dashboards and settings and all other indices.
+* `ro`: some write requests can go through to the `.kibana` index so that UI state in discover can be saved and short urls can be created.
+* `rw`: some more actions will be allowed towards the `.kibana` index only, so Kibana dashboards and settings can be modified.
+* `admin`: like above, but has additional permissions to use the ReadonlyREST PRO/Enterprise Kibana app
+* `unrestricted`: when no `kibana_access` rule is set, no action is restricted. This option is equivalent of no `kibana_access` rule used. 
+
+**NB:** The "admin" access level does not mean the user will be allowed to access all indices/actions. It's just like "rw" with settings changes privileges. If you want really unrestricted access for your Kibana user, including ReadonlyREST PRO/Enterprise app, set `kibana_access: unrestricted`. You can use this rule with the `users` rule to restrict access to selected admins.
+
+This rule is often used with the `indices` rule, to limit the data a user is able to see represented on the dashboards. In that case do not forget to allow the custom kibana index in the `indices` rule!
+
+#### `kibana_index`
+
+`kibana_index: .kibana-user1`
 
 **Default value is `.kibana`**
 
 Specify to what index we expect Kibana to attempt to read/write its settings \(use this together with `kibana.index` setting in kibana.yml.\)
 
-This value directly affects how `kibana_access` works because at all the access levels \(yes, even admin\), `kibana_access` rule will **not** maatch any _write_ request in indices that are not the designated kibana index.
+This value directly affects how `kibana_access` works because at all the access levels \(yes, even admin\), `kibana_access` rule will **not** match any _write_ request in indices that are not the designated kibana index.
 
 If used in conjunction with ReadonlyREST Enterprise, this rule enables **multi tenancy**, because in ReadonlyREST, a tenancy is identified with a set of Kibana configurations, which are by design collected inside a kibana index \(default: `.kibana`\).
 
+#### `kibana_index_template`
+
+`kibana_index_template: .kibana_template`
+
+[TODO:]
```

Comment:
@sscarduzio @Dzuming  could you please help here?

Reply:
Used to pre-populate tenancies with default kibana objects, like dashboards and visualizations. Thus providing a starting point for new tenants that will avoid the bad user experience of logging for the first time and finding a completely empty Kibana.